playground/eShopLite/CatalogDb/Program.cs

```diff
@@ -1,9 +1,24 @@
 using CatalogDb;
+using Microsoft.EntityFrameworkCore;
+using Npgsql;
 
 var builder = WebApplication.CreateBuilder(args);
 
 builder.AddServiceDefaults();
-builder.AddNpgsqlDbContext<CatalogDbContext>("catalogdb");
+
+// Configure Npgsql.EntityFrameworkCore.PostgreSQL services
+if (builder.Configuration.GetConnectionString("catalogdb") is string connectionString)
```

Comment:
What's the purpose of the guard here? It seems odd that if there's no connection string we don't add the context, but we call the Aspire method on line 21 anyway.

Reply:
the `connectionString` argument in `AddNpgsqlDataSource` is non-nullable so it doesn't build if we don't check. So it's really for the build to succeed more than guarding null values, the sample doesn't work with a null connection string.

Is a bang (!) better? DbContext doesn't use the connection string so putting in the block is not better but semantically.

Reply:
`!` definitely not better. I think a nicer pattern is to verify the connection string has a value and throw if not, before doing anything else, similar to the sample in my other comment.

Reply:
Fixed